tests/test_waku_netconfig.nim

```diff
@@ -0,0 +1,351 @@
+{.used.}
+
+import
+  chronos,
+  libp2p/multiaddress,
+  testutils/unittests
+
+import
+    ../../waku/node/config,
+    ../../waku/waku_enr/capabilities,
+    ../../apps/wakunode2/external_config
+
+
+proc defaultTestWakuFlags(): CapabilitiesBitfield =
+  CapabilitiesBitfield.init(
+      lightpush = false,
+      filter = false,
+      store = false,
+      relay = true
+  )
+    
+suite "Waku NetConfig":
+  
+  asyncTest "Create NetConfig with default values":
+
+    let conf = defaultTestWakuNodeConf()
+    
+    let wakuFlags = defaultTestWakuFlags()
+    
+    let netConfigRes = NetConfig.init(
+      bindIp = conf.listenAddress,
+      bindPort = conf.tcpPort,
+      extIp = none(ValidIpAddress),
+      extPort = none(Port),
+      extMultiAddrs = @[],
+      wsBindPort = conf.websocketPort,
+      wsEnabled = conf.websocketSupport,
+      wssEnabled = conf.websocketSecureSupport,
+      dns4DomainName = none(string),
+      discv5UdpPort = none(Port),
+      wakuFlags = some(wakuFlags)
+    )
+
+    check:
+      netConfigRes.isOk()
+
+  asyncTest "AnnouncedAddresses contains only bind address when no external addresses are provided":
+
+    let conf = defaultTestWakuNodeConf()
+        
+    let netConfigRes = NetConfig.init(
+      bindIp = conf.listenAddress,
+      bindPort = conf.tcpPort
+    )
+     
+    require:
+      netConfigRes.isOk()
```

Comment:
I'd always apply the next technique for such cases:

```suggestion
    assert netConfigRes.isOk(), $netConfigRes.error
```

That applies elsewhere.

Reply:
Agree that it's a better technique! Done :)